deploy/charts/rook-ceph-cluster/templates/cephblockpool.yaml

```diff
@@ -1,53 +1,49 @@
 {{- $root := . -}}
-{{- range $blockpool := .Values.cephBlockPools -}}
+{{- range $blockpool := .Values.cephBlockPools }}
 ---
-apiVersion: ceph.rook.io/v1
 kind: CephBlockPool
+apiVersion: ceph.rook.io/v1
 metadata:
   name: {{ $blockpool.name }}
   namespace: {{ $root.Release.Namespace }} # namespace:cluster
 spec:
-{{ toYaml $blockpool.spec | indent 2 }}
+  {{- $blockpool.spec | toYaml | nindent 2 }}
+{{- if $blockpool.storageClass.enabled }}
 ---
-{{- if default false $blockpool.storageClass.enabled }}
-apiVersion: storage.k8s.io/v1
 kind: StorageClass
+apiVersion: storage.k8s.io/v1
 metadata:
-  name: {{ $blockpool.storageClass.name }}
-{{- if $blockpool.storageClass.labels }}
+  name: {{ $blockpool.storageClass.name | quote }}
```

Comment:
If the storageclass name would look like `"123"`, the rendered template would look like `name: 123`, and be considered a number - which in turn would make k8s api-server react and say its not a valid StorageClass specification.

Due to this, one needs `| quote` here and in similar places. I have not ensured this consistently in this PR.